src/event-stream/event-server.ts

```diff
@@ -96,8 +96,21 @@ async function handleMempoolTxsMessage(rawTxs: string[], db: DataStore): Promise
   await db.updateMempoolTxs({ mempoolTxs: dbMempoolTxs });
 }
 
-async function handleClientMessage(msg: CoreNodeBlockMessage, db: DataStore): Promise<void> {
-  const parsedMsg = parseMessageTransactions(msg);
+async function handleClientMessage(
+  chainId: ChainID,
+  msg: CoreNodeBlockMessage,
+  db: DataStore
+): Promise<void> {
+  const parsedMsg = parseMessageTransactions(chainId, msg);
+
+  // Delete on-btc-chain transactions that failed, there is no useful data in them,
+  // and the db and API schemas can't currently fit these types of transactions.
+  for (let i = parsedMsg.transactions.length - 1; i >= 0; i--) {
+    if (!parsedMsg.parsed_transactions[i]) {
+      parsedMsg.parsed_transactions.splice(i, 1);
+      parsedMsg.transactions.splice(i, 1);
+    }
+  }
```

Comment:
There is no way for these transactions to have an event, right? Because otherwise some downstream code will throw, and progress will halt.

Reply:
From Aaron earlier:
> I'm pretty sure that tx receipts will be generated even for bitcoin operations that fail (and therefore don't have events)

So probably okay? 🤷